lahja/base.py

```diff
@@ -395,6 +390,20 @@ def get_connected_endpoints_and_subscriptions(
         """
         ...
 
+    @property
+    @abstractmethod
+    def is_local_broadcast_enabled(self) -> bool:
```

Comment:
I'm not sold on this API to be honest. If we just look at the history of the project, it is true that in the previous version you had to explicitly connect to your own endpoint to be able to act upon your own events. But looking further back this features has actually existed from day one *without* any explicit API (that changed when we moved away from the central router architecture to the mesh architecture).

But history aside, do we really gain much from making this opt-in? The only pro argument that I see is a (probably tiny) performance improvement for not having to check the subscribed events if the feature is disabled. Other than that, performance should not play a key role here as we don't process local events if we aren't locally subscribed.

I would also argue that it makes the APIs feel less coherent. E.g I'm not sure if `True` is the right answer to `is_connected_to('myself')`. Plus, having it either return `True` or `False` depending on the fact if `local_broadcast` is enabled or not doesn't feel right to me. Nothing changes in terms of "being connected". 

In my mind `is_connected_to('myself')` could simply always throw an exception stating something like *"Endpoints do not hold connections to themselves"* (Relying on you native speakers to round the edges but you get the idea).

I **do want** to be able to say `is_any_remote_subscribed_to(SomeEvent)` and have that take our own subscriptions into account but I think that we should rename that API to `is_any_endpoint_subscribed_to` to make that more coherent. *Any* **endpoint**  does include our self but *any* **remote**  does not sound like an API that should factor in our own subscriptions.

I'm not saying *over my dead body* but I would like to raise these concerns and also hear @lithp opinion on this.

Reply:
I think the two awkward parts for me that keep me from being immediately :+1: about making this just the default *baseline* behavior are `is_connected_to` and `are_all_remotes_subscribed` to.

`is_connected_to(<myself>)` returning `True` unconditionally could be fine, just needs to be well documented.  Not a big hang-up.  That naturally must then extend to `get_connected_remotes_and_subscriptions` including `<myself>` which is a bit more weird, but again, not that big a deal.

The big one for me is `are_all_remotes_subscribed_to(...)`.  I feel like it is very un-ergonomic to require an endpoint to subscribe to the thing itself for this to return true, since it *seems* like the common case is some central coordinator that is waiting for everyone else to subscribe.  **BUT** I'm not sure that use case is actually real.  In `trinity` I think that we'd need the *self* endpoint to also be subscribed so maybe my arguments are moot..... (maybe I just talked myself out of it).

Curious also to hear @lithp opinion and I think I'm *"ok"* with just making this the default behavior.